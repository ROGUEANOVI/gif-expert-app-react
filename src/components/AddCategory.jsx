import { useState } from 'react';

export const AddCategory = ({ onAddCategory }) => {

    const [inputValue, setInputValue] = useState('');

    const onInputChange = (e) => {
        setInputValue(e.target.value);
    };

    const onSubmit = (e) => {
        e.preventDefault();
        if (inputValue.trim().length <= 1) return;
    
        onAddCategory(inputValue.trim());
        setInputValue('');
    };
    
    return (
        <>
            <form aria-label='form' onSubmit={ onSubmit }>
                <input 
                    type="text" value={ inputValue } 
                    onChange={ onInputChange } 
                    placeholder="Buscar gifs" 
                />
            </form>
        </>
    );
};
