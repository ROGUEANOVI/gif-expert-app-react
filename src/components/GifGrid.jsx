import { useFetchGifs } from '../hooks/useFetchGifs';
import { GifGridItem } from './GifGridItem';

export const GifGrid = ({ category }) => {

    const {gifs, isLoading} = useFetchGifs(category);

    return (
        <>
            <h3>{ category }</h3>
            <hr />

            {isLoading && <h2>Loading...</h2>}
            
            
            <div className="card-grid">
                {
                    gifs.map((gif) => (
                        <GifGridItem key={ gif.id } { ...gif } />
                    ))
                }   
            </div>
        </>
    );
};
