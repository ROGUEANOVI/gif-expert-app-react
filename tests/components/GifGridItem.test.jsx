import { render, screen } from "@testing-library/react";
import { GifGridItem } from "../../src/components/GifGridItem";

describe('Test <GifGridItem />', () => { 
    
    const title = 'test';
    const url = 'https://localhost/test.jpg';

    test('should match snapshot', () => {
        
        const { container } = render(<GifGridItem title={ title } url={ url } />);
        
        expect(container).toMatchSnapshot();
    });

    test('should image with url and alt', () => {

        render(<GifGridItem title='test' url={ url } />);
        
        const { src, alt } = screen.getByRole('img');

        expect(src).toBe(url);
        expect(alt).toBe(title);
    });

    test('should display the title', () => {

        render(<GifGridItem title={ title } url={ url } />);
        
        expect(screen.getByText(title)).toBeTruthy();
    });
});