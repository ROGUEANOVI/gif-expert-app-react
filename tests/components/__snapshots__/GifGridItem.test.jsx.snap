// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test <GifGridItem /> should match snapshot 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="test"
      src="https://localhost/test.jpg"
    />
    <p>
      test
    </p>
  </div>
</div>
`;
